### ISSUE TITLE
OS Font lookup does not work in Flatpak (at least on some distros)

### ISSUE THREAD

**Issue by dweymouth** [flathub/io.github.dweymouth.supersonic#104]
See https://github.com/dweymouth/supersonic/issues/500#issuecomment-2501575545 for more info.

Flatpak sandboxing is preventing Supersonic from being able to lookup fonts from the OS to display Japanese characters or other characters that are not in the bundled font that comes with the app. It may be as simple as adding an additional permission to the Flatpak manifest but I have no idea.

**Comment by anarcat** [comment]
that's strange: i don't have that problem and also often have weird characters in my filenames (my native language is french) so I'm pretty sure i would have noticed.

@thecivilizedgamer could you expand on your desktop environment here? do you have `xdg-desktop-portal-gtk` installed?

similar issues https://github.com/flatpak/flatpak/issues/5389 https://github.com/flatpak/flatpak/issues/4571

the TLDR is that this should work, your system fonts should be available inside the flatpak.

**Comment by thecivilizedgamer** [comment]
> that's strange: i don't have that problem and also often have weird characters in my filenames (my native language is french) so I'm pretty sure i would have noticed.
> 
> @thecivilizedgamer could you expand on your desktop environment here? do you have `xdg-desktop-portal-gtk` installed?
> 
> similar issues [flatpak/flatpak#5389](https://github.com/flatpak/flatpak/issues/5389) [flatpak/flatpak#4571](https://github.com/flatpak/flatpak/issues/4571)
> 
> the TLDR is that this should work, your system fonts should be available inside the flatpak.

Yep, I do have `xdg-desktop-portal-gtk` installed. I'm running Xubuntu 24.04.1 LTS, and it's a pretty fresh install - I've only installed a few applications, and haven't really done anything to the OS. I can provide additional details if you want.

As a side note, I noticed that the Flatpak version was prompting me for the password for my Nextcloud Music server on every launch, but the natively installed version does not seem to have that problem. Maybe that's tangentially related, I can file a separate issue for it if you want.

**Comment by s-fleck** [comment]
I have the same problem on xubuntu und kubuntu. as a sidenode, "weird" european characters are not an issue, but japanese etc don't work

**Comment by dweymouth** [comment]
@s-fleck Have you tried the direct installation or the new AppImage? I believe this is only a Flatpak problem.

**Comment by s-fleck** [comment]
Haven't tried it yet, but it would be great to get it working in flatpak. It also seems all the font dirs are mounted correctly - If i go to settings/experimental/application-fonts and browse run/host directory, all expected font directories show up (.e.g as documented here: https://docs.flatpak.org/en/latest/desktop-integration.html). 

Could it be supersonic is not looking at all these flatpak-specific directories for finding the fonts? i'm not really sure how all this works

**Comment by dweymouth** [comment]
I don't know much about how Flatpak sandboxing works. But this part of the source code of the `go-text/typesetting` package may be helpful. It appears it looks for fonts in /etc/fonts or the path specified in the `$FONTCONFIG_PATH` env var

**Comment by anarcat** [comment]
> I don't know much about how Flatpak sandboxing works. But [this part](https://github.com/go-text/typesetting/blob/main/fontscan/fontconfig.go) of the source code of the `go-text/typesetting` package may be helpful. It appears it looks for fonts in /etc/fonts or the path specified in the `$FONTCONFIG_PATH` env var

@s-fleck there was pointing at the [flatpak docs](https://docs.flatpak.org/en/latest/desktop-integration.html#fonts) which say:

> # Fonts
> 
> Flatpak exposes fonts from the host to the sandbox, and the runtime ships the default fontconfig from upstream.
> 
> Fonts from `/usr/share/fonts` are exposed at `/run/host/fonts`, from `/usr/local/share/fonts` at `/run/host/local-fonts`, and from `$XDG_DATA_HOME/fonts` at `/run/host/user-fonts`.
> 
> `~/.fonts` is a legacy path and should not be used.
> 
> Fontconfig config files from the host are not exposed.

I don't know how "normal" apps find their fonts, but I can only assume either FONTCONFIG_PATH is set by flatpak or those apps also look in /run,,,

**Comment by jcharaoui** [comment]
Same issue here, arabic characters aren't rendered properly using the Flatpak:

![Image](https://github.com/user-attachments/assets/4883a90c-1407-49ab-973a-41139df722e2)

This doesn't happen when I examine the files in my file manager or with other software.

**Comment by jcharaoui** [comment]
Similar issue: https://github.com/flathub/com.valvesoftware.Steam/issues/1070
